src/sdk/wxscintilla/include/wx/wxscintilla.h

```diff
@@ -44,25 +44,21 @@
 #include <wx/versioninfo.h>
 #endif
 
-// this workarounds an issue that in some build configuration of wx library, WXEXPORT is defined as empty,
-// as WXDLLIMPEXP_SCI is defined as WXEXPORT below, this may cause wxScintilla class not be exported from
-// the codeblocks.dll
-#ifdef __WXMSW__
-    #ifdef WXEXPORT
-        #undef WXEXPORT
-        #define WXEXPORT __declspec(dllexport)
-    #else // WXEXPORT not defined
-        #define WXEXPORT __declspec(dllexport)
-    #endif // WXEXPORT
-#endif // __WXMSW__
-
 #ifdef WXMAKINGDLL_SCI
-    #define WXDLLIMPEXP_SCI WXEXPORT
+    // this workarounds an issue that in some build configuration of wx library, WXEXPORT is defined as empty,
+    // as WXDLLIMPEXP_SCI is defined as WXEXPORT below, this may cause wxScintilla class not be exported from
+    // the codeblocks.dll
+    #ifdef __WXMSW__
```

Comment:
Actually are you sure this is really required? It seems that for the versions of GCC we use this is correctly defined, see here https://github.com/wxWidgets/wxWidgets/blob/master/include/wx/dlimpexp.h#L37 It seems this has been changed in 3.1 and the fix is 5 years old: https://github.com/wxWidgets/wxWidgets/commit/db966da330a1caf634df8f96dd78640b7d9695fb I don't think you should be bothered by problems in 3.0.

Reply:
Yes, I know this commit, indeed this issue is fixed.
This hunk of my patch just handle the wx release before this commit.
So, we can simply skip this patch.

Reply:
Yes, building with 3.0 on Windows is not supported. There is no point adding complexity for very little benefit.

Reply:
Remove this offending commit and then push the change to svn...